docs/adr/002-policy-on-using-caret-with-dependencies.md

```diff
@@ -0,0 +1,87 @@
+# ADR 002: Policy on Using Caret (`^`) or tilde (`~`) with Dependencies in `package.json`
+
+## Status
+Proposed
+
+## Submitters
+- Ulises Gascón (@UlisesGascon)
+
+## Decision Owners
+- Express TC (@expressjs/express-tc)
+
+## Context
+Historically, the Express project has avoided using the caret (`^`) in the `package.json` files for its own dependencies. This decision aims to review whether this practice should continue or if adjustments are needed.
+
+**Why do we need this decision?**  
+Clarifying the policy on using caret (`^`) helps to ensure consistency across the Express ecosystem, reduce unnecessary maintenance, and prevent unexpected dependency updates. It also addresses concerns about the balance between keeping dependencies up-to-date and avoiding regressions.
+
+**What problem does it solve or avoid?**  
+This decision aims to avoid unexpected updates and regressions from external dependencies while reducing the maintenance burden of frequently updating pinned dependencies.
+
+**Are there any existing issues/discussions/pull requests related to this?**  
+- [Discussion: Using caret (^) with our own dependencies #279](https://github.com/expressjs/discussions/issues/279)
+- [expressjs/express#6017 (comment)](https://github.com/expressjs/express/issues/6017)
+
+## Decision
+We will adopt a policy where the caret (`^`) symbol is used for dependencies owned within the Express ecosystem (e.g., `body-parser` for Express), but continue to avoid using it for third-party dependencies that we do not maintain.
+
+**What will be done?**  
+- Update the `package.json` files to use `^` for our own dependencies.
+- Continue using fixed versions (no caret or tilde) for external dependencies to minimize risks of unintended updates.
+
+**Note on `^` vs. `~`:**  
+- `^` allows updates to the most recent minor or patch version, offering greater flexibility and reducing the need for frequent manual updates. For example, `^1.2.3` will accept updates to `1.3.0`, `1.4.0`, but not `2.0.0`.
+- `~` is more conservative, only allowing updates to patch versions. For example, `~1.2.3` will accept updates to `1.2.4`, `1.2.5`, but not `1.3.0`.
+- For our own dependencies that adhere strictly to semver, `^` is preferred over `~`, while for third-party libraries, a stricter versioning strategy is maintained to prevent unexpected regressions.
+
+**What will not be done?**  
+- We will not use `^` or `~` for external dependencies that are not maintained by the Express organization, as they could introduce unexpected changes.
+- We won't force to use `^` or `~` for own dependencies if there is a reason to use a fix version and it is properly documented.
+
+## Rationale
+
+**Alternatives Considered:**
+- **Alternative 1:** Use `^` for all dependencies, including third-party libraries.  
+- **Reason for rejection:** This could increase the risk of unintended regressions and security issues from third-party updates.
```

Comment:
imo this risk is lower than the risk of dependency vulnerability/regression *fixes* from *not* being available to downstream users.